packages/core/src/index.ts

```diff
@@ -0,0 +1,180 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable @typescript-eslint/require-await */
+
+export enum RouteType {
+  GMP = "gmp",
+  FUNGIBLE = "fungible",
+}
+
+export enum Environment {
+  LOCAL = "local",
+  DEVNET = "devnet",
+  TESTNET = "testnet",
+  MAINNET = "mainnet",
+}
+
+export enum FeeHandlerType {
+  BASIC = "basic",
+  PERCENTAGE = "percentage",
+  UNDEFINED = "undefined",
+}
+
+export enum Network {
+  EVM = "evm",
+  SUBSTRATE = "substrate",
+}
+
+export type Domain = {
+  sygmaId: number;
+  // https://chainagnostic.org/CAIPs/caip-2
+  caipId: string;
+  chainId: number;
+  name: string;
+  iconUrl?: string;
+  type: Network;
+};
+
+export type Resource = EvmResource | SubstrateResource;
+
+export enum ResourceType {
+  FUNGIBLE = "fungible",
+  NON_FUNGIBLE = "nonfungible",
+  PERMISSIONED_GENERIC = "permissionedGeneric",
+  PERMISSIONLESS_GENERIC = "permissionlessGeneric",
+}
+
+interface BaseResource {
+  sygmaResourceId: string;
+  // https://chainagnostic.org/CAIPs/caip-19
+  caip19: string;
+  type: ResourceType;
+  iconUrl?: string;
+  native?: boolean;
+  burnable?: boolean;
+  symbol?: string;
+  decimals?: number;
+}
+
+export type EvmResource = BaseResource & {
+  address: string;
+};
+
+export type XcmMultiAssetIdType = {
+  concrete: {
+    parents: number;
+    interior:
+      | "here"
+      | {
+          x3: Array<{ parachain: number } | { generalKey: [number, string] }>; // This is a tuple of length and value
+        };
+  };
+};
+
+export type SubstrateResource = BaseResource & {
+  assetID?: number;
+  assetName: string;
+  xcmMultiAssetId: XcmMultiAssetIdType;
+};
+
+export type Route = {
+  fromDomain: Domain;
+  toDomain: Domain;
+  resource: Resource;
+  feeHandler: FeeHandler;
+};
+
+export type TransferStatus = "pending" | "executed" | "failed";
+
+export type TransferStatusResponse = {
```

Comment:
Should this also return the details of the transactions (sourceHash, destinationHash...)?

Reply:
I would love to add it but afaik indexer doesn't have that in response?! cc @MakMuftic

Reply:
I'm pretty sure we return the whole `Transfer` object from the db so it should all be there already.

Reply:
Maybe somebody can share openapi link for indexer? :)